stdlib/pathlib/_abc.pyi

```diff
@@ -0,0 +1,192 @@
+    def open(
+        self, mode: OpenBinaryMode, buffering: Literal[0], encoding: None = None, errors: None = None, newline: None = None
+    ) -> FileIO: ...
+    # Buffering is on: return BufferedRandom, BufferedReader, or BufferedWriter
+    @overload
+    def open(
+        self,
+        mode: OpenBinaryModeUpdating,
+        buffering: Literal[-1, 1] = -1,
+        encoding: None = None,
+        errors: None = None,
+        newline: None = None,
+    ) -> BufferedRandom: ...
+    @overload
+    def open(
+        self,
+        mode: OpenBinaryModeWriting,
+        buffering: Literal[-1, 1] = -1,
+        encoding: None = None,
+        errors: None = None,
+        newline: None = None,
+    ) -> BufferedWriter: ...
+    @overload
+    def open(
+        self,
+        mode: OpenBinaryModeReading,
+        buffering: Literal[-1, 1] = -1,
+        encoding: None = None,
+        errors: None = None,
+        newline: None = None,
+    ) -> BufferedReader: ...
+    # Buffering cannot be determined: fall back to BinaryIO
+    @overload
+    def open(
+        self, mode: OpenBinaryMode, buffering: int = -1, encoding: None = None, errors: None = None, newline: None = None
+    ) -> BinaryIO: ...
+    # Fallback if mode is not specified
+    @overload
+    def open(
+        self, mode: str, buffering: int = -1, encoding: str | None = None, errors: str | None = None, newline: str | None = None
+    ) -> IO[Any]: ...
+    def read_bytes(self) -> bytes: ...
+    def read_text(self, encoding: str | None = None, errors: str | None = None, newline: str | None = None) -> str: ...
+    def write_bytes(self, data: ReadableBuffer) -> int: ...
+    def write_text(
+        self, data: str, encoding: str | None = None, errors: str | None = None, newline: str | None = None
+    ) -> int: ...
+    def iterdir(self) -> Generator[Self, None, None]: ...
+    def glob(
+        self, pattern: str, *, case_sensitive: bool | None = None, recurse_symlinks: bool = True
+    ) -> Generator[Self, None, None]: ...
+    def rglob(
+        self, pattern: str, *, case_sensitive: bool | None = None, recurse_symlinks: bool = True
+    ) -> Generator[Self, None, None]: ...
+    def walk(
+        self, top_down: bool = True, on_error: Callable[[OSError], object] | None = None, follow_symlinks: bool = False
+    ) -> Iterator[tuple[Self, list[str], list[str]]]: ...
+    def absolute(self) -> Self: ...
+    @classmethod
+    def cwd(cls) -> Self: ...
+    def expanduser(self) -> Self: ...
+    @classmethod
+    def home(cls) -> Self: ...
+    def readlink(self) -> Self: ...
+    def resolve(self, strict: bool = False) -> Self: ...
+    def symlink_to(self, target: StrOrBytesPath, target_is_directory: bool = False) -> None: ...
+    def hardlink_to(self, target: StrOrBytesPath) -> None: ...
+    def touch(self, mode: int = 0o666, exist_ok: bool = True) -> None: ...
+    def mkdir(self, mode: int = 0o777, parents: bool = False, exist_ok: bool = False) -> None: ...
```

Comment:
Looks like [`copy`](https://github.com/python/cpython/blob/d0abd0b826cfa574d1515c6f8459c9901939388f/Lib/pathlib/_abc.py#L867) and [`copy_into`](https://github.com/python/cpython/blob/d0abd0b826cfa574d1515c6f8459c9901939388f/Lib/pathlib/_abc.py#L895) are missing:
```suggestion
    def mkdir(self, mode: int = 0o777, parents: bool = False, exist_ok: bool = False) -> None: ...
    def copy(
        self, target: StrPath, *, follow_symlinks: bool = True, dirs_exist_ok: bool = False, preserve_metadata: bool = False
    ) -> StrPath: ...
    def copy_into(
        self, target_dir: StrPath, *, follow_symlinks: bool = True, dirs_exist_ok: bool = False, preserve_metadata: bool = False
    ) -> StrPath: ...
```